romancal/tweakreg/tests/test_astrometric_utils.py

```diff
@@ -406,3 +434,49 @@ def test_get_catalog_using_invalid_parameters(ra, dec, sr, catalog_name):
         get_catalog(ra, dec, sr=sr, catalog=catalog_name)
 
     assert str(exec_info.typename).lower().endswith("error")
+
+
+@pytest.mark.parametrize(
+    "ra, dec, epoch, sr, catalog_name",
+    [
+        (10, 10, 2000, 0.1, "GAIADR1"),
+        (10, 10, 2010.3, 0.1, "GAIADR2"),
+        (10, 10, 2030, 0.1, "GAIADR3"),
+        (10, -10, 2000, 0.1, "GAIADR1"),
+        (10, -10, 2010.3, 0.1, "GAIADR2"),
+        (10, -10, 2030, 0.1, "GAIADR3"),
+        (0, 0, 2000, 0.01, "GAIADR1"),
+        (0, 0, 2010.3, 0.01, "GAIADR2"),
+        (0, 0, 2030, 0.01, "GAIADR3"),
+    ],
+)
+def test_get_catalog_using_epoch(ra, dec, epoch, sr, catalog_name):
+    """Test that get_catalog returns coordinates corrected by proper motion."""
+
+    result = get_catalog(ra, dec, epoch=epoch, sr=sr, catalog=catalog_name)
+    returned_ra = np.array(result["ra"])
+    returned_dec = np.array(result["dec"])
+
+    # get GAIA data and update coords to requested epoch using pm measurements
+    gaia_ref_epoch = 2016.0
+    gaia_ref_epoch_coords = get_catalog(
+        ra, dec, epoch=gaia_ref_epoch, sr=sr, catalog=catalog_name
+    )
+    expected_new_ra = (
+        np.array(
+            gaia_ref_epoch_coords["ra"] * 3600
+            + (epoch - gaia_ref_epoch) * gaia_ref_epoch_coords["pmra"] / 1000
+        )
+        / 3600
+    )
```

Comment:
I am pretty sure that the pmra's are defined with a factor of cos(dec) so that np.hypot(pmra, pmdec) is the speed of the star's motion on the sky.  This test doesn't have that factor, so I'm surprised it passes?  OTOH, np.isclose is not so stringent when it comes to astrometry, where an arcsecond of motion is a lot but is only 5e-6 radians.

The parallax correction is being neglected in these tests, which will also lead to some expected error.  I'm a bit surprised that there are _no_ nearby stars that are triggering errors there?  Is there a reason that the (0, 0) search radii are different than the 10, 10 and 10, -10 search radii?

Reply:
Thanks for catching the missing spherical correction term in the equation, Eddie.

Regarding the search radii, there's no specific reason for the different values for (ra, dec) = (0, 0). I had copied those parameters from another test and forgot to change them to better suit the new test.

Reply:
We added a more strict condition for `np.isclose()`: `atol=1e-10` and `rtol=1e-9`; these values are used in the following equation to determine if two values are close: `absolute(a - b) <= (atol + rtol * absolute(b))` (from [numpy's docs](https://numpy.org/doc/stable/reference/generated/numpy.isclose.html)).